src/Integration/Binding/SolutionRuleSetWriter.cs

```diff
@@ -66,15 +65,15 @@ public string WriteSolutionLevelRuleSet(string solutionFullPath, RuleSet downloa
         /// Generate a solution level rule set file path from the given <see cref="ProjectInformation"/>.
         /// </summary>
         /// <param name="ruleSetRootPath">Root directory to generate the full file path under</param>
-        /// <param name="sonarQubeProject">SonarQube project to generate a rule set file name path for</param>
+        /// <param name="sonarQubeProjectKey">SonarQube project key to generate a rule set file name path for</param>
         /// <param name="fileNameSuffix">Fixed file name suffix</param>
-        internal /* testing purposes */ static string GenerateSolutionRuleSetPath(string ruleSetRootPath, ProjectInformation sonarQubeProject, string fileNameSuffix)
+        internal /* testing purposes */ static string GenerateSolutionRuleSetPath(string ruleSetRootPath, string sonarQubeProjectKey, string fileNameSuffix)
         {
             Debug.Assert(fileNameSuffix != null);
 
             // Cannot use Path.ChangeExtension here because if the sonar project name contains
             // a dot (.) then everything after this will be replaced with .ruleset
-            string fileName = $"{PathHelper.EscapeFileName(sonarQubeProject.Key + fileNameSuffix)}.{FileExtension}";
+            string fileName = $"{PathHelper.EscapeFileName(sonarQubeProjectKey + fileNameSuffix)}.{FileExtension}";
```

Comment:
The `fileNameSuffix` here is just appended to the SonarQube project key. It's always looked quite strange/awkward IMO when you get solution-level rulesets that are named like "roslynCSharp.ruleset", rather than something a bit nicer like "roslyn.CSharp.ruleset". I think it's easier to read by keeping the project key and the language separate 'more separate', especially for project keys which are long/complicated/will contain the project branch.
Could also change the value passed in for fileNameSuffix to include a dot "." or dash "-", or do it here.